### PR TITLE
add rubocop checkstyle

### PR DESCRIPTION
**Pull request by huanzhang**
In order to make the code to be more maintainable by community people, like other popular ruby projects, we should use rubocop as the code analyzer and run it every time before we make a  pull request.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-source 'https://rubygems.org'
+source 'https://ruby.taobao.org'
```
Comment (plribeiro3000) [inline Gemfile]: Can you leave the original rubygems?

It is more standard and should work for everyone. Is there a particular reason why you changed this?

File: fog-aliyun.gemspec [hunk 2]
```diff
@@ -4,24 +4,25 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
...
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
-  spec.require_paths = ["lib"]
+  spec.require_paths = ['lib']
 
-  spec.add_development_dependency "bundler", "~> 1.10"
-  spec.add_development_dependency "rake", "~> 10.0"
-  spec.add_development_dependency "rspec", "~> 3.3"
+  spec.add_development_dependency 'bundler', '~> 1.10'
+  spec.add_development_dependency 'rake', '~> 10.0'
+  spec.add_development_dependency 'rspec', '~> 3.3'
```
Comment (plribeiro3000) [inline fog-aliyun.gemspec]: The standard in the `fog` community is to use `minitest`.

Can you switch to minitest plz?

File: Gemfile (hunk 1 above)
Comment (huanzhang) [inline Gemfile]: @plribeiro3000 I change it as I am in China and the GFW blocks rubygems.org :(
But I forgot to change it back.

I will change it back shortly!

File: lib/fog/aliyun.rb [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
-require "fog/json"
-require "fog/aliyun/version"
+require 'fog/core'
+require 'fog/json'
+require 'fog/aliyun/version'
 
 module Fog
   module Compute
-    ret = autoload :Aliyun, "fog/aliyun/compute"
+    ret = autoload :Aliyun, 'fog/aliyun/compute'
```
Comment (plribeiro3000) [inline lib/fog/aliyun.rb]: You don't need the `ret =` part here.

File: Gemfile (hunk 1 above)
Comment (plribeiro3000) [inline Gemfile]: Dammm it. Did not know it is blocked. =s

File: fog-aliyun.gemspec (hunk 2 above)
Comment (huanzhang) [inline fog-aliyun.gemspec]: Actually, we don't have a real unit test. We can change the test framework when adding real test.

File: lib/fog/aliyun.rb (hunk 3 above)
Comment (huanzhang) [inline lib/fog/aliyun.rb]: I agree, but how about if we change this in another commit?

File: fog-aliyun.gemspec (hunk 2 above)
Comment (plribeiro3000) [inline fog-aliyun.gemspec]: Sure.

File: lib/fog/aliyun.rb (hunk 3 above)
Comment (plribeiro3000) [inline lib/fog/aliyun.rb]: sure.

File: Gemfile (hunk 1 above)
Comment (plribeiro3000) [inline Gemfile]: Can you switch this value back?

Then i will merge it right away.

**Comment by huanzhang** [comment]
@plribeiro3000 
please merge this pr if you feel comfortable :)
I'm going to add some more feature based on those commit.

**Comment by plribeiro3000** [comment]
Done. Thank you for your contribution!

**Comment by huanzhang** [comment]
thanks @plribeiro3000

**Pull request merged**